### ISSUE TITLE
Static files localization in RCL

### ISSUE THREAD

**Issue by meancrazy** [aspnet/AspNetCore#3471]
RCL with static files with "localizable" names (like `messages.en.js`) throws 500 error.

What happens: in RCL static files are included as embedded resources. And compiler includes localized files in separate dlls. And when you try to get them, you get an error:
```
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/2.0 GET http://localhost/Test/js/messages.en.js
fail: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[1]
      An unhandled exception has occurred while executing the request.
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Extensions.FileProviders.Embedded.Manifest.ManifestFileInfo.EnsureLength()
   at Microsoft.Extensions.FileProviders.Embedded.Manifest.ManifestFileInfo.get_Length()
   at Microsoft.AspNetCore.StaticFiles.StaticFileContext.LookupFileInfo()
   at Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

Sample project with tests: https://github.com/meancrazy/LocalizationInRCL/

**Comment by pranavkm** [comment]
cc @javiercn

**Comment by Eilon** [comment]
Ah, RCL = Razor Class Libraries?

**Comment by Eilon** [comment]
Just to clarify, did you *want* to serve the file in the RCL using the StaticFiles middleware? Or you got this error by accident?

**Comment by meancrazy** [comment]
So the sample project (https://github.com/meancrazy/LocalizationInRCL/) has two tests. First with `test.js` passed and second with `messages.en.js` fails

**Comment by javiercn** [comment]
@meancrazy You can exclude those with 
```
<ItemGroup>
  <EmbeddedResource Update="*.js" ExcludeFromManifest="true" />
<ItemGroup>
```

**Comment by javiercn** [comment]
@meancrazy Thanks for contacting us.

There are two issues happening here. One is in your sample, you have to setup the culture for the request.
The other one is in the manifest embedded file provider. We don't do anything special to support localization scenarios. What we could do is:

Call `Assembly.GetManifestResourceStream` and if that returns null, call `Assembly.GetSatelliteAssembly(CultureInfo.CurrentCulture)` and then call `GetManifestResourceStream` on the result.

For that to work, the current culture needs to be setup appropriately.

**Comment by meancrazy** [comment]
@javiercn as for me – do not localize (in dlls) any static files 😄js libs should do it as yourself

**Comment by meancrazy** [comment]
For example: in devextreme (https://unpkg.com/devextreme@18.1.5/dist/js/localization/) localization will be changed on the fly by calling
```js
DevExpress.localization.locale("ru");
```

**Comment by javiercn** [comment]
We could actually do what I described above but with the culture coming from the manifest, that way we are not bound by having to setup the culture in the context appropriately.

**Comment by cdroulers** [comment]
I'm experiencing the same issue.

If I rename my file to `dx.messages_fr.js` the problem goes away and the file is successfully returned.

**Comment by shirhatti** [comment]
@mkArtakMSFT @javiercn Is this planned for 3.0?

**Comment by javiercn** [comment]
The recommendation here is to use the static web assets feature that shipped in 3.0. Closing this issue as there's no more action.
